core-rust/core-api-server/src/core_api/server.rs

```diff
@@ -78,13 +78,15 @@ use state_manager::jni::state_manager::ActualStateManager;
 use super::{constants::LARGE_REQUEST_MAX_BYTES, handlers::*, not_found_error, ResponseError};
 
 use handle_status_network_configuration as handle_provide_info_at_root_path;
+use state_manager::simple_mempool::SimpleMempool;
 use state_manager::store::StateManagerDatabase;
 
 #[derive(Clone)]
 pub(crate) struct CoreApiState {
     pub network: NetworkDefinition,
     pub state_manager: Arc<RwLock<ActualStateManager>>,
     pub database: Arc<RwLock<StateManagerDatabase>>,
+    pub mempool: Arc<RwLock<SimpleMempool>>,
```

Comment:
_(a note from the author)_

So, @malexandru-rdx , taking https://github.com/radixdlt/babylon-node/pull/411#pullrequestreview-1390194356 into account - is this the moment to start pushing a `CoreApiState` instance across the stack? (other ideas welcome too)

Reply:
I personally like to see verbose code when setting up different connected "services" and also I'd rather read it as a flatter config file than having multiple layers of abstraction.
In this case, other than passing `CoreApiState` directly to `create_server` I would not do anything else (not sure about this either).